### PR TITLE
Trigger for L2 tests (#9)

### PR DESCRIPTION
**Pull request by rgsubh**
* Trigger for L2 tests

* Testing evnts

* Testing events for this commit

* Testing events after variable name change in L2 repo.

* Modified to triger only for master and releases branch

* Updating repo name

* Testing trigger on repo

* Testing for demo

* FOR DEMO WILL BE REVERTED

* REVERTED

* FOR DEMO WILL BE REVERTED

* REVERTED

### CODE REVIEW

File: .github/workflows/intergation-tests.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - 'releases/*'
+jobs: 
+    trigger-integration-tests:
+      name: Trigger Integration tests
+      runs-on: ubuntu-latest
+      steps:
+        - name: Extract branch name
+          id: extract_branch
+          run: |
+            echo "##[set-output name=branchname;]$(echo ${GITHUB_REF##*/})"
```
Comment (ds-ms) [inline .github/workflows/intergation-tests.yml:19]: This is probably not required right?

You could use `${{ env.GITHUB_REF }}` in the condition?

File: .github/workflows/intergation-tests.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        - name: Trigger Test run
+          if: github.event.pull_request.base.ref == 'releases/v1' || steps.extract_branch.outputs.branchname == 'releases/v1' || github.event.pull_request.base.ref == 'master' || steps.extract_branch.outputs.branchname == 'master'
+          run: |
+           response=$(curl -X POST -H "Authorization: token ${{ secrets.L2_REPO_TOKEN }}" https://api.github.com/repos/Azure/azure-actions-integration-tests/dispatches --data '{"event_type": "K8sCreateSecretReleasesPR", "client_payload": {"action": "CreateSecret", "commit": "${{ github.event.pull_request.head.sha }}", "repository":  "${{ github.repository }}", "prNumber": "${{ github.event.pull_request.number }}", "tobranch": "${{ github.event.pull_request.base.ref }}",  "frombranch": "${{ github.event.pull_request.head.ref }}" }}')
```
Comment (ds-ms) [inline .github/workflows/intergation-tests.yml:24]: I don't like the entire inline thing.. it's hard to read.

File: .github/workflows/intergation-tests.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      name: Trigger Integration tests
+      runs-on: ubuntu-latest
+      steps:
+        - name: Extract branch name
+          id: extract_branch
+          run: |
+            echo "##[set-output name=branchname;]$(echo ${GITHUB_REF##*/})"
+
+        - name: Trigger Test run
+          if: github.event.pull_request.base.ref == 'releases/v1' || steps.extract_branch.outputs.branchname == 'releases/v1' || github.event.pull_request.base.ref == 'master' || steps.extract_branch.outputs.branchname == 'master'
```
Comment (ds-ms) [inline .github/workflows/intergation-tests.yml:22]: Given that you have kept a script for extra script for fetching the branchname.. you could just move the entire condition logic to the script.
Add something like `##[set-output name=valid_target_branch;](true || false)`

File: .github/workflows/intergation-tests.yml (hunk 1 above)
Comment (rgsubh) [reply .github/workflows/intergation-tests.yml:24]: This is required as  ${{ env.GITHIB_REF }} gives the full path, the name needs to be extracted

File: .github/workflows/intergation-tests.yml (hunk 3 above)
Comment (rgsubh) [reply .github/workflows/intergation-tests.yml]: The current version must be more readable.

File: .github/workflows/intergation-tests.yml (hunk 2 above)
Comment (rgsubh) [reply .github/workflows/intergation-tests.yml]: Moved to a file.

**Comment by ds-ms** [comment]
Pinging @rgsubh

**Pull request merged**